### PR TITLE
Unit overhaul & Optional APEX weapon pack

### PR DESCRIPTION
**Pull request by TyroneMF**
- Overhauled units clothing/gear
- Added optics to some weapons
- Final fix on Tan mags showing on Black MX rifles
- Added additional contractor names to CfgWorlds
- Renamed "Specialist" to "Machine Gunner" (Classname unchanged)
- Added new UI images for editorPreviews.

### CODE REVIEW

**Comment by TyroneMF** [comment]
The optional weapons pack was designed to replace the current HLC one we use.

File: optionals/weapons_apex/config.cpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        requiredAddons[] = {"tacs_units"};
+        author = ECSTRING(main,Author);
+        authors[] = {"Tyrone, Lairdd1989"};
+        url = ECSTRING(main,URL);
+        VERSION_CONFIG;
+    };
+};
+
+#include "CfgWeapons.hpp"
+#include "CfgVehicles.hpp"
```
Comment (jonpas) [inline optionals/weapons_apex/config.cpp]:
```suggestion
#include "CfgVehicles.hpp"

```

File: optionals/weapons_apex/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# About
+
+Replaces unit's vanilla weapons with APEX weapons.
```
Comment (jonpas) [inline optionals/weapons_apex/README.md]:
```suggestion
Replaces unit's base Arma 3 weapons with Arma 3 Apex weapons.
```

File: optionals/weapons_apex/CfgWeapons.hpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    class arifle_SPAR_01_blk_F;
+    class CLASS(tac_HK416_Mod): arifle_SPAR_01_blk_F {
+        class LinkedItems {
+            class LinkedItemsOptic {
+                slot = "CowsSlot";
+                item = "optic_holosight_blk_f";
+            };
+        };
+    };
+};
```
Comment (jonpas) [inline optionals/weapons_apex/CfgWeapons.hpp]:
```suggestion
};

```

File: optionals/weapons_apex/CfgWeapons.hpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
+class CfgWeapons {
+    class arifle_MXM_Black_F;
+    class CLASS(tac_MXM_Mod): arifle_MXM_Black_F {
```
Comment (jonpas) [inline optionals/weapons_apex/CfgWeapons.hpp]: Why does this have a `tac` midfix? `CLASS(x)` -> `tacs_x` already.

File: addons/units/CfgVehicles_PublicUnits.hpp [hunk 5]
```diff
@@ -194,27 +204,26 @@ class CLASS(Unit_I_Specialist): CLASS(Unit_Combat_RS_BS_GP_BB) {
...
     icon = "iconManMG";
 
     headgearList[] = {
-        QCLASS(Cap_BlackLogo), 1,
-        QCLASS(Cap_TanLogo), 1,
-        QCLASS(Cap_Earpiece_BlackLogo), 1,
-        QCLASS(Cap_Earpiece_TanLogo), 1,
-        QCLASS(Cap_Backwards_BlackLogo), 1,
-        QCLASS(Cap_Backwards_TanLogo), 1
+        "H_HelmetB", 1,
```
Comment (jonpas) [inline addons/units/CfgVehicles_PublicUnits.hpp]: Put vanilla ones at the end for organization. Try to sort mod ones same as they are defined in the mod.

Comment (TyroneMF) [reply addons/units/CfgVehicles_PublicUnits.hpp]: Done

File: optionals/weapons_apex/CfgWeapons.hpp (hunk 4 above)
Comment (TyroneMF) [reply optionals/weapons_apex/CfgWeapons.hpp]: Fixed, was an oversight.

**Comment by jonpas** [comment]
Need to check image generation.

File: addons/units/CfgVehicles_PublicUnits.hpp [hunk 6]
```diff
@@ -18,7 +18,9 @@ class CLASS(Unit_I_Bodyguard): CLASS(Unit_Polo_CP_LS_TP_OB) {
     };
 
     CBA_facewearList[] = {
-        "G_Shades_Black", 1
+        "G_Shades_Black", 0.30,
+        "G_Aviator", 0.30,
+        "G_Spectacles_Tinted", 0.30,
```
Comment (jonpas) [inline addons/units/CfgVehicles_PublicUnits.hpp]:
```suggestion
        "G_Spectacles_Tinted", 0.30
```

File: optionals/weapons_apex/CfgWeapons.hpp [hunk 7]
```diff
@@ -0,0 +1,41 @@
+class CfgWeapons {
+    class arifle_MXM_Black_F;
+    class CLASS(arifle_MXM_Mod): arifle_MXM_Black_F {
```
Comment (jonpas) [inline optionals/weapons_apex/CfgWeapons.hpp]:
```suggestion
    class CLASS(MXM_Black_HAMR): arifle_MXM_Black_F {
```

File: addons/weapons/CfgWeapons.hpp [hunk 8]
```diff
@@ -60,4 +60,26 @@ class CfgWeapons {
...
+
+        class LinkedItems {
+            class LinkedItemsOptics {
+                slot = "CowsSlot";
+                item = "optic_aco";
+            };
+        };
+    };
+
+    class CLASS(MX_BlackCamo_GL_ACO): CLASS(MX_GL_BlackCamo) {
```
Comment (jonpas) [inline addons/weapons/CfgWeapons.hpp]:
```suggestion
    class CLASS(MX_GL_BlackCamo_ACO): CLASS(MX_GL_BlackCamo) {
```

File: optionals/weapons_apex/CfgWeapons.hpp [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        class LinkedItems {
+            class LinkedItemsOptic {
+                slot = "CowsSlot";
+                item = "optic_hamr";
+            };
+        };
+    };
+
+    class arifle_SPAR_01_GL_blk_F;
+    class CLASS(arifle_HK416GL_Mod): arifle_SPAR_01_GL_blk_F {
```
Comment (jonpas) [inline optionals/weapons_apex/CfgWeapons.hpp]:
```suggestion
    class CLASS(HK416_GL_Black_HAMR): arifle_SPAR_01_GL_blk_F {
```

File: optionals/weapons_apex/CfgWeapons.hpp [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+        class LinkedItems {
+            class LinkedItemsOptic {
+                slot = "CowsSlot";
+                item = "optic_hamr";
+            };
+        };
+    };
+
+    class LMG_03_F;
+    class CLASS(arifle_FN_Mod): LMG_03_F {
```
Comment (jonpas) [inline optionals/weapons_apex/CfgWeapons.hpp]:
```suggestion
    class CLASS(M249_Holo): LMG_03_F {
```

File: optionals/weapons_apex/CfgWeapons.hpp [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+        class LinkedItems {
+            class LinkedItemsOptic {
+                slot = "CowsSlot";
+                item = "optic_holosight_blk_f";
+            };
+        };
+    };
+
+    class arifle_SPAR_01_blk_F;
+    class CLASS(arifle_HK416_Mod): arifle_SPAR_01_blk_F {
```
Comment (jonpas) [inline optionals/weapons_apex/CfgWeapons.hpp]:
```suggestion
    class CLASS(HK416_Black_Holo): arifle_SPAR_01_blk_F {
```

File: optionals/weapons_apex/CfgWeapons.hpp [hunk 12]
```diff
@@ -0,0 +1,41 @@
+class CfgWeapons {
+    class arifle_MXM_Black_F;
+    class CLASS(arifle_MXM_Mod): arifle_MXM_Black_F {
+        class LinkedItems {
```
Comment (jonpas) [inline optionals/weapons_apex/CfgWeapons.hpp]: Missing `scope = 1;` like in core preset weapons.

File: addons/units/CfgVehicles_PublicUnits.hpp [hunk 13]
```diff
@@ -67,31 +70,32 @@ class CLASS(Unit_B_Contractor): CLASS(Unit_I_Contractor) {
...
     author = "Pomigit, Jonpas";
     displayName = CSTRING(Unit_Contractor_GL);
     editorPreview = QPATHTOF(UI\Unit_Contractor_GL.jpg);
-    weapons[] = {QCLASS(MX_GL_BlackCamo), "hgun_P07_F", "Binocular", "Throw", "Put"};
-    respawnWeapons[] = {QCLASS(MX_BlackCamo), "hgun_P07_F", "Binocular", "Throw", "Put"};
-    magazines[] = {ITEMS_5(30Rnd_65x39_caseless_mag), "1Rnd_SmokeBlue_Grenade_shell", "1Rnd_SmokeGreen_Grenade_shell", "1Rnd_SmokeRed_Grenade_shell", ITEMS_3(1Rnd_HE_Grenade_shell), ITEMS_2(16Rnd_9x21_Mag), "SmokeShell", "SmokeShellGreen", ITEMS_2(HandGrenade)};
-    respawnMagazines[] = {ITEMS_5(30Rnd_65x39_caseless_mag), "1Rnd_SmokeBlue_Grenade_shell", "1Rnd_SmokeGreen_Grenade_shell", "1Rnd_SmokeRed_Grenade_shell", ITEMS_3(1Rnd_HE_Grenade_shell), ITEMS_2(16Rnd_9x21_Mag), "SmokeShell", "SmokeShellGreen", ITEMS_2(HandGrenade)};
-    linkedItems[] = {DEFAULT_ITEMS, QCLASS(Cap_BlackLogo), "G_Shades_Black", QCLASS(Vest_Tactical_DarkBlack)};
-    respawnLinkedItems[] = {DEFAULT_ITEMS, QCLASS(Cap_BlackLogo), "G_Shades_Black", QCLASS(Vest_Tactical_DarkBlack)};
+    weapons[] = {QCLASS(MX_BlackCamo_GL_ACO), "hgun_P07_F", "Binocular", "Throw", "Put"};
```
Comment (jonpas) [inline addons/units/CfgVehicles_PublicUnits.hpp]: Error, wrong class.

**Pull request merged**